File: example/fizzbuzz/fizz_buzz.hs
func :: Integer -> [String]
func 0 = []
func n 
    | mod n 15 == 0 = func(n-1) ++ ["FizzBuzz"]
    | mod n 5 == 0 = func(n-1) ++ ["Buzz"]
    | mod n 3 == 0 = func(n-1) ++ ["Fizz"]
    | otherwise = func(n-1) ++ [show n]

main = do n <- readLn
          print (func n)

File: example/fizzbuzz/fizz_buzz.rb
for n in 1..ARGV[0].to_i
  if n % 15 == 0
    print "FizzBuzz "
  elsif n % 5 == 0
    print "Buzz "
  elsif n % 3 == 0
    print "Fizz "
  else
    print n.to_s + " "
  end
end

